pkg/opavalidation/opavalidation.go

```diff
@@ -0,0 +1,427 @@
+	if err != nil {
+		return nil, err
+	}
+	actionItems, err := actionItemsFromRegoResult(regoResult)
+	if err != nil {
+		return actionItems, err
+	}
+	err = updateActionItemsWithObjectFields(actionItems, objectAsMap)
+	if err != nil {
+		return actionItems, err
+	}
+	updateActionItemsWithEventType(actionItems, eventType)
+	return actionItems, nil
+}
+
+// runRegoForObject executes rego with a Kubernetes object as input.
+func runRegoForObject(ctx context.Context, regoAsString string, object map[string]interface{}) (rego.ResultSet, error) {
+	query, err := rego.New(rego.EnablePrintStatements(true), rego.PrintHook(topdown.NewPrintHook(os.Stdout)),
+		rego.Query("results = data"),
+		rego.Module("fairwinds", regoAsString),
+		rego.Function2(
+			&rego.Function{
+				Name: "kubernetes",
+				Decl: types.NewFunction(types.Args(types.S, types.S), types.A),
+			},
+			func(_ rego.BuiltinContext, _ *ast.Term, _ *ast.Term) (*ast.Term, error) {
+				// Perhaps do something mroe here to communicate it isn't possible to fetch
+				// cluster resources?
+				return nil, nil
+			},
+		),
+		rego.Function1(
+			&rego.Function{
+				Name: "insightsinfo",
+				Decl: types.NewFunction(types.Args(types.S), types.A),
+			},
+			validateInsightsInfoFunctionArgs())).PrepareForEval(ctx)
+	if err != nil {
+		return nil, err
+	}
+	preparedInput := rego.EvalInput(object)
+	rs, err := query.Eval(ctx, preparedInput)
+	if err != nil {
+		return nil, err
+	}
+	return rs, nil
+}
+
+// validateInsightsInfoFunctionArgs validates the argument that would be
+// passed to our rego InsightsInfo function without executing code provided by
+// the real InsightsInfo function used elsewhere in Insights.
+// This function returns a function that is called from rego.Function().
+func validateInsightsInfoFunctionArgs() func(rego.BuiltinContext, *ast.Term) (*ast.Term, error) {
+	return func(bc rego.BuiltinContext, inf *ast.Term) (*ast.Term, error) {
+		reqInfo, err := getStringFromAST(inf)
+		if err != nil {
+			return nil, rego.NewHaltError(fmt.Errorf("unable to convert requested InsightsInfo to string: %w", err))
+		}
+		switch strings.ToLower(reqInfo) {
+		case "context", "cluster":
+			return nil, nil
+		/* Potentially return a value to the policy instead of nil?
+		   retInfoAsValue, err := ast.InterfaceToValue(retInfo)
+		   		if err != nil {
+		   			return nil, rego.NewHaltError(fmt.Errorf("unable to convert information %q to ast value: %w", retInfo, err))
+		   		}
+		   		return ast.NewTerm(retInfoAsValue), nil
+		*/
+		default:
+			return nil, rego.NewHaltError(fmt.Errorf("cannot return unknown Insights Info %q", reqInfo))
+		}
+	}
+}
+
+// getStringFromAST converts the ast.Term type used by the rego pkg, into a
+// string.
+func getStringFromAST(astTerm *ast.Term) (string, error) {
+	astString, ok := astTerm.Value.(ast.String)
+	if !ok {
+		return "", errors.New("expected a string")
+	}
+	return strings.Trim(astString.String(), "\""), nil
```

Comment:
why do you have to trim the `"`?

Reply:
The AST String function explicitly quotes the value before returning it, but we don't want literal quotes in our string.